homeassistant/components/overkiz/coordinator.py

```diff
@@ -132,9 +134,12 @@ async def _async_update_data(self) -> dict[str, Device]:
             elif event.name == EventName.DEVICE_STATE_CHANGED:
                 for state in event.device_states:
                     device = self.devices[event.device_url]
-                    if state.name not in device.states:
-                        device.states[state.name] = state
-                    device.states[state.name].value = self._get_state(state)
+
+                    if (device_state := device.states[state.name]) is None:
+                        device_state = state
+                        device.states[state.name] = device_state
```

Comment:
Why changed this part of the code? What was wrong ?

Reply:
mypy didn't like this. In the previous situation `device.states[state.name]` can return None, due to 
`def __getitem__(self, name: str) -> State | None`. This is now handled.